hadoop-hdfs-project/hadoop-hdfs/src/main/resources/hdfs-default.xml

```diff
@@ -4440,8 +4440,29 @@
   <value>3000</value>
   <description>
     Base time window in ms for DFSClient retries.  For each retry attempt,
-    this value is extended linearly (e.g. 3000 ms for first attempt and
-    first retry, 6000 ms for second retry, 9000 ms for third retry, etc.).
+    this value is extended exponentially based on dfs.client.retry.window.multiplier.
+  </description>
+</property>
+
+<property>
+  <name>dfs.client.retry.window.multiplier</name>
+  <value>1</value>
+  <description>
+    Multiplier for extending the retry time window.  For each retry attempt,
+    the retry time window is extended by multiplying dfs.client.retry.window.base
+    by this multiplier raised to the power of the current failure count. The default
+    value of 1 means the window will expand linearly (e.g. 3000 ms for first attempt
+    and first retry, 6000 ms for second retry, 9000 ms for third retry, etc.).
+  </description>
+</property>
+
+<property>
+  <name>dfs.client.retry.window.max</name>
+  <value>2147483647</value>
```

Comment:
The default value for `dfs.client.retry.window.max` is too high here. In some corner case, it will sleep very long time?

Reply:
Thanks so much for the review @Hexiaoqiao. The reason I chose this value was that I wanted the changes in this PR to be totally transparent to existing users -- so the backoff should work exactly as it does today for anyone who upgrades. I don't know how people have tuned their backoffs today, so adding a lower max might affect their configured backoffs. The default case will be well-bounded by the default retries of 3. That said, I agree that there's very little utility in waiting many minutes on a backoff. What if I put this to 30s?

Was this the only concern in terms of the default action? My test case [testDefaultRetryPolicy](https://github.com/apache/hadoop/pull/3271/files#diff-ce7234e2a352097945c5348774888466b48f9574d4764d9415559d975d380884R47) proves that the default case remains unchanged from trunk. The default case was determined based on the [comment in DFSInputStream](https://github.com/apache/hadoop/pull/3271/files#diff-f6cd7fbb9a909e1679c2d77df54b0b2fe661a66c8c89fb455fde313966d51a61L1018-L1026), the old implementation details, and my own testing of the backoff policy prior to this change.

I also created this spreadsheet that helped me to determine how different multiplier values might affect the backoff: https://docs.google.com/spreadsheets/d/1I9ejqDtJ6-krSh-YBt0qHTf3JwZu5zRlrOhbzY0kJAg/edit?usp=sharing